src/seedu/addressbook/parser/Parser.java

```diff
@@ -238,16 +241,36 @@ private int parseArgsAsDisplayedIndex(String args) throws ParseException, Number
      * @return the prepared command
      */
     private Command prepareFind(String args) {
-        final Matcher matcher = KEYWORDS_ARGS_FORMAT.matcher(args.trim());
-        if (!matcher.matches()) {
+        final Matcher matcherTagsOnly = TAGS_ARGS_FORMAT.matcher(args.trim());
+        final Matcher matcherNamesOnly = NAMES_ARGS_FORMAT.matcher(args.trim());
+        Matcher matcher;
+        boolean isFindTag = false;
+
+        if (!matcherNamesOnly.matches() && !matcherTagsOnly.matches()) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     FindCommand.MESSAGE_USAGE));
         }
+        if (matcherTagsOnly.matches()) {
+            matcher = matcherTagsOnly;
+            isFindTag = true;
+        } else {
+            matcher = matcherNamesOnly;
+        }
```

Comment:
It doesn't seem like you can search for name and tags at the same time?

Reply:
Yes that is right. I'll will work on changing this.